### PR TITLE
[SPIKE] swapping busboy for multer

### PR DESCRIPTION
**Pull request by ckousik**
```
    new file: core/server/middleware/ghost-multer.js 
modified:   core/server/middleware/index.js
modified:   core/server/routes/api.js
modified:   package.json
```

Multer works fine when I run this on localhost

### CODE REVIEW

File: core/server/middleware/index.js [hunk 1]
```diff
@@ -34,6 +35,7 @@ var bodyParser      = require('body-parser'),
 
 middleware = {
     busboy: busboy,
```
Comment (acburdine) [inline core/server/middleware/index.js]: Wouldn't this line & the other related busboy code be removed?

File: package.json [hunk 2]
```diff
@@ -52,6 +52,7 @@
     "lodash": "3.10.1",
     "moment": "2.11.1",
     "morgan": "1.6.1",
+    "multer": "^1.1.0",
```
Comment (acburdine) [inline package.json]: This should be locked to a specific version: aka remove the `^`

**Comment by acburdine** [comment]
Hey @ckousik! Thanks for contributing!

From a quick glance, this looks good! (Haven't tested it in practice though).

Couple comments (excluding the other ones I made):
- can you change the commit message to this: `deps: multer@1.1.0`
- the reason tests are failing is you have some linting errors: if you run `grunt lint` locally it will show you your errors
- if this truly removes busboy, all busboy-related code and references should be deleted
- it would be cool if you could try and write some sort of tests (not sure how easy it is)

**Comment by jtwebman** [comment]
I don't see any file changes besides including the dependency?

**Comment by ckousik** [comment]
I actually messed up when squashing commits....and now I don't know how to fix it. There's supposed to be quite a few changes

**Comment by ErisDS** [comment]
Time to get familiar with the reflog!

`git reflog -20` will show you the last 20 states that git knows about:



Here you can see that each commit gets a `HEAD@{#}` reference. So if I wanted to go back before my rebase, I would need `HEAD@{3}`... you can see the next command would be `git reset HEAD@{3}`.

`git reset HEAD@{#}` will reset you back to the corresponding state.

This should bring your changes back

**Comment by ErisDS** [comment]
FYI - this PR seems to have 2 commits still.

I have some other feedback but it's more relevant to the discussion I think, so I'll write it on the issue.

**Comment by ErisDS** [comment]
https://github.com/TryGhost/Ghost/issues/6362#issuecomment-179309604

If we do press ahead with this change, the i18n keys for busboy should also be removed, or they should be changed to generic file upload keys and the equivalent error handling should be done.

**Comment by ErisDS** [comment]
I left some comments on the issue and there has been further discussion but not yet any further discussion around the implementation in this PR. Going to close this for now - it is linked to the issue so we can use it for discussion still, but it needs some more discussion & updates before it could be merged.

**Pull request closed without merging**